script.js

```diff
@@ -0,0 +1,141 @@
+                    getVid(videoDetails.items)
+                });
+        });
+//     fetch("./search.json")
+//         .then(searchResults => searchResults.json())
+//         .then(searchResults => {
+//             let ids = []
+//             for (let video of searchResults.items) {
+//                 ids.push(video.id.videoId);
+//             }
+//             fetch("./searchvide.json")
+//                 .then(videoDetails => videoDetails.json())
+//                 .then(videoDetails => {
+//                     for (let i in searchResults.items)
+//                         videoDetails.items[i].snippet.shortDesc = searchResults.items[i].snippet.description;
+//                     getVid(videoDetails.items)
+//                 });
+//         });
+ });
+
+
+
+
+
+
+function getVid(videoDetails) {
+    let i = 1
+    for (let video of videoDetails) {
+        fillDetails(video, i++);
+    }
+    hideAll();
+    displayVideos(1)
+}
+
+window.onresize = function () {
+    // console.log(document.querySelectorAll(".active")[0].getAttribute("id"));
+    displayVideos(document.querySelectorAll(".active")[0].getAttribute("id").substring(4));
+}
+
+function fillDetails(details, i) {
+    // console.log(details);
+    let video = document.importNode(template, true);
+    video.querySelector(".ytVideo").setAttribute("id", "video" + i);
+    video.querySelector(".title").textContent = details.snippet.title;
+    video.querySelector(".views").textContent = details.statistics.viewCount + " views";
+    video.querySelector(".published").textContent = details.snippet.publishedAt.substring(0, 10).split("-").join("/");
+    video.querySelector(".author").textContent = details.snippet.channelTitle;
+    video.querySelector(".author").setAttribute("onclick", "window.open('https://www.youtube.com/channel/" + details.snippet.channelId + "')");
+    video.querySelector(".description").textContent = details.snippet.shortdescription;
+
+
+    // video.querySelector(".vid").setAttribute("src", "https://www.youtube.com/embed/" + details.id);
+    video.querySelector(".thumbnail").src = details.snippet.thumbnails.high.url;
+    video.querySelector(".thumbnail").setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    // video.querySelector("#video" + i).setAttribute("onclick", "window.open('https://www.youtube.com/watch?v=" + details.id + "')");
+
+    document.querySelector(".main").appendChild(video);
+}
+
+
+function hideAll() {
+    for (let i = 1; i <= 15; i++) {
+        document.querySelector("#video" + i).style.display = "none";
+    }
+}
+
+
+
+function generatePages(count) {
+    let pages = document.querySelector(".pages");
+    pages.textContent = "";
+    for (let i = 1; i <= count; i++) {
+        let page = document.createElement("div");
+        page.setAttribute("class", "page");
+        let btn = document.createElement("button");
```

Comment:
can we create a new function that is responsible to get the page with its metadata?